Mathlib/AlgebraicGeometry/Morphisms/Proper.lean

```diff
@@ -0,0 +1,69 @@
+/-
+Copyright (c) 2024 Christian Merten, Andrew Yang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Christian Merten, Andrew Yang
+-/
+import Mathlib.AlgebraicGeometry.Morphisms.Separated
+import Mathlib.AlgebraicGeometry.Morphisms.UniversallyClosed
+import Mathlib.AlgebraicGeometry.Morphisms.FiniteType
+
+/-!
+
+# Proper morphisms
+
+A morphism of schemes is proper if it is separated, universally closed and (locally) of finite type.
+Note that we don't require quasi-compact, since this is implied by universally closed (TODO).
```

Comment:
Have you formalised this in another repository already? If so, I believe it is ok to the definition as is.
Otherwise, I believe you should at least add a pointer to this https://mathoverflow.net/questions/23337/is-a-universally-closed-morphism-of-schemes-quasi-compact (I have not looked at the details yet, but I would give some trust the author of the answer...)

Reply:
It is in the valuative criterion project.

Reply:
More specifically: Here https://github.com/leanprover-community/mathlib4/blob/742d3b40eefe44b43da330ed7fc55d19dc82d50c/Mathlib/ValuativeCriterion/UniversallyClosed.lean#L450 And it is exactly the proof you linked to above.

Reply:
(And I essentially followed that proof by Poonen)